File: TIMS-X.Server/TIMS-X.Server/JsonModels/UserGroup.ts




    export class UserGroup { 
        public name: string;
        public inactive: boolean;
        public protected: boolean;
        public description: string;
        public settings: UserGroupAppSetting[];
        public updatedUserId: number;
        public updatedDate: Date;
        public userReferences: UserGroupReference[];
		
        public pendingDelete: boolean;
        public id: number;
        public hasStateBeenSet: boolean;
        
    }

File: TIMS-X.Core/Domain/UserGroup.cs
using System;
using System.Collections.Generic;
using TIMS_X.Core.Domain.Base;

namespace TIMS_X.Core.Domain
{
    public class UserGroup : Entity, IUpdateAudited
    {
        public UserGroup()
        {
            UserReferences = new HashSet<UserGroupReference>();
        }

        public string Name { get; set; }
        public bool Inactive { get; set; }
        public bool Protected { get; set; }
        public string Description { get; set; }

        public ICollection<UserGroupAppSetting> Settings { get; set; }
        public int? UpdatedUserId { get; set; }
        public DateTime UpdatedDate { get; set; }
        public ICollection<UserGroupReference> UserReferences { get; set; }
    }
}